akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -37,6 +37,17 @@ trait ReplicatedData {
 
 }
 
+trait DeltaReplicatedData extends ReplicatedData {
+  type D <: ReplicatedData
+
+  def delta: Option[D]
+
+  def mergeDelta(d: D): T
+
+  def clearDelta: T
```

Comment:
I've always imagined the methods delta and clearDelta to be implemented on the level of the Replicator. The Delta-CRDT implementation can simply be an immutable data-structure that returns a delta on all of its mutations. The replicator actor can then have full control on what delta to dispatch where (one might want to send different deltas to different peers)